File: frontend/src/api/evaluation.ts
import apiClient from './client'
import type { CheckResultStatus, TargetType } from '@/types'

export interface CreateSessionDto {
  name: string
  target_type: TargetType
  target_url?: string
  target_description?: string
  template_id: string
}

export interface CheckResult {
  id: string
  session_id: string
  check_item_id: string
  status: CheckResultStatus
  notes?: string
  evidence?: string
  raw_output?: string
  last_poc_output?: string
  actual_severity?: string
  confidence_score?: number
  confidence_level?: 'high' | 'medium' | 'low' | 'unknown'
}

export interface PocTaskStatus {
  task_id: string
  task_state: string
  session_id: string
  check_item_id: string
  result_status: CheckResultStatus
  last_poc_output?: string
  confidence_score?: number
  confidence_level?: 'high' | 'medium' | 'low' | 'unknown'
  diagnosis_code?: string
  diagnosis_message?: string
  exit_code?: number
  message?: string
}

export interface EvaluationSession {
  id: string
  name: string
  target_type: TargetType
  target_url?: string
  target_description?: string
  score?: number
  status: 'draft' | 'in_progress' | 'completed' | 'archived'
  template_id: string
  created_at: string
  results?: CheckResult[]
}

export const evaluationApi = {
  getSessions: (): Promise<EvaluationSession[]> => {
    return apiClient.get('/evaluations')
  },

  getSession: (id: string): Promise<EvaluationSession> => {
    return apiClient.get(`/evaluations/${id}`)
  },

  createSession: (data: CreateSessionDto): Promise<EvaluationSession> => {
    return apiClient.post('/evaluations', data)
  },

  updateResult: (
    sessionId: string,
    checkItemId: string,
    data: Partial<CheckResult>
  ): Promise<{ message: string }> => {
    return apiClient.patch(`/evaluations/${sessionId}/results/${checkItemId}`, data)
  },

  runPoc: (
    sessionId: string,
    checkItemId: string
  ): Promise<{ status: string; task_id?: string; message: string }> => {
    return apiClient.post(`/evaluations/${sessionId}/results/${checkItemId}/run-poc`)
  },

  getPocTaskStatus: (
    sessionId: string,
    checkItemId: string,
    taskId: string
  ): Promise<PocTaskStatus> => {
    return apiClient.get(`/evaluations/${sessionId}/results/${checkItemId}/poc-task/${taskId}`)
  },

  exportReportHtml: (sessionId: string): Promise<Blob> => {
    return apiClient.get(`/report/${sessionId}/report`, {
      responseType: 'blob',
      headers: { Accept: 'text/html' },
    })
  },

  updateSession: (id: string, data: Partial<EvaluationSession>): Promise<EvaluationSession> => {
    return apiClient.patch(`/evaluations/${id}`, data)
  },

  deleteSession: (id: string): Promise<{ message: string }> => {
    return apiClient.delete(`/evaluations/${id}`)
  },
}

File: backend/schema/evaluation.py
"""
评估会话相关的 Pydantic 验证模型
"""
import uuid
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field

from model.checklist import RiskLevel, TargetType
from model.evaluation import CheckResultStatus, SessionStatus
from schema.checklist import ChecklistTemplateResponse


class CheckResultBase(BaseModel):
    status: CheckResultStatus = CheckResultStatus.PENDING
    actual_severity: Optional[RiskLevel] = None
    raw_output: Optional[str] = None
    evidence: Optional[str] = None
    last_poc_output: Optional[str] = None
    notes: Optional[str] = None


class CheckResultUpdate(CheckResultBase):
    pass


class CheckResultResponse(CheckResultBase):
    id: uuid.UUID
    session_id: uuid.UUID
    check_item_id: uuid.UUID
    checked_by_id: Optional[uuid.UUID] = None
    checked_at: Optional[datetime] = None
    confidence_score: Optional[int] = None
    confidence_level: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class PocRunResponse(BaseModel):
    status: str
    task_id: Optional[str] = None
    message: str


class PocTaskStatusResponse(BaseModel):
    task_id: str
    task_state: str
    session_id: uuid.UUID
    check_item_id: uuid.UUID
    result_status: CheckResultStatus
    last_poc_output: Optional[str] = None
    confidence_score: Optional[int] = None
    confidence_level: Optional[str] = None
    diagnosis_code: Optional[str] = None
    diagnosis_message: Optional[str] = None
    exit_code: Optional[int] = None
    message: Optional[str] = None


class EvaluationSessionBase(BaseModel):
    name: str = Field(..., title="评估任务名称")
    target_type: TargetType
    target_url: Optional[str] = None
    target_description: Optional[str] = None
    template_id: uuid.UUID
    status: SessionStatus = SessionStatus.DRAFT
    notes: Optional[str] = None


class EvaluationSessionCreate(BaseModel):
    name: str
    target_type: TargetType
    target_url: Optional[str] = None
    target_description: Optional[str] = None
    template_id: uuid.UUID


class EvaluationSessionResponse(EvaluationSessionBase):
    id: uuid.UUID
    assignee_id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    completed_at: Optional[datetime] = None
    score: Optional[float] = None

    model_config = ConfigDict(from_attributes=True)


class EvaluationSessionDetailResponse(EvaluationSessionResponse):
    """包含完整结果和关联模板的详情"""
    results: List[CheckResultResponse] = []
    # 也可以把完整的 template 带回来，但为了轻量化，通常拆分获取，或者这里直接附带
